### PR TITLE
feat+refactor: Improved and extended `Skeleton.subskel()`

### PR DESCRIPTION
**Pull request by phorward**
This PR introduces a new way to define sub-skeletons, additionally to the old one.

1. Renames `Skeleton.subSkel()` into `Skeleton.subskel()` with a deprecation warning
2. Giving names as parameter merges the bones of all `Skeleton.subSkels`-configurations together. This is the usual behavior.
3. Given the `bones`-parameter allows to freely specify a sub-skeleton; One specialty here is,  that the order of the bones can also be changed in this mode!
4. 2 + 3 can be combined, but then the original order of the bones is kept
5. Mostly backward-compatible to the original `Skeleton.subSkel()` except that `fnmatch` is used to resolve wildcards, which is more powerful (the old implementation only accepted "*" at the end of the name).

Example Skeleton:
```py
from viur.core import skeleton, utils
from viur.core.bones import *


class TodoSkel(skeleton.Skeleton):
    subSkels = {
        "*": ["*name"],
        "add": ["subject"],
    }

    firstname = StringBone(
        descr="Vorname",
    )

    lastname = StringBone(
        descr="Nachname",
        required=True,
    )

    category = SelectBone(
        descr="Kategorie",
        defaultValue="question",
        required=True,
        values={
            "question": "Anfrage",
            "billing": "Abrechnung",
            "service": "Service",
        },
    )

    subject = StringBone(
        descr="Thema",
        required=True,
    )

    message = TextBone(
        descr="Nachricht",
        required=True,
        validHtml=None,
    )


    status = SelectBone(
        descr="Status",
        required=True,
        defaultValue="open",
        values={
            "open": "Offen",
            "pending": "In Bearbeitung",
            "closed": "Geschlossen",
        },
    )

    user = UserBone(
         descr="Zugewiesen an",
    )
```

Example usage:
```py
# legacy mode (see 2)
subskel = TodoSkel.subskel("add")
# creates subskel: key, firstname, lastname, subject

# free mode (see 3):
subskel = TodoSkel.subskel(bones=("subject", "message", "*stname"))
# creates subskel: key, subject, message, firstname, lastname

# mixed mode (see 4)
subskel = TodoSkel.subskel("add", bones=("message", ))
# creates subskel: key, firstname, lastname, subject, message
```

### CODE REVIEW

**Comment by phorward** [comment]
Example:

## skeletons/todo.py
```py
from viur.core import skeleton, utils
from viur.core.bones import *


class TodoSkel(skeleton.Skeleton):
    subSkels = {
        "*": ["*name"],
        "add": ["subject"],
    }

    firstname = StringBone(
        descr="Vorname",
    )

    lastname = StringBone(
        descr="Nachname",
        required=True,
    )

    category = SelectBone(
        descr="Kategorie",
        defaultValue="question",
        required=True,
        values={
            "question": "Anfrage",
            "billing": "Abrechnung",
            "service": "Service",
        },
    )

    subject = StringBone(
        descr="Thema",
        required=True,
    )

    message = TextBone(
        descr="Nachricht",
        required=True,
        validHtml=None,
    )


    status = SelectBone(
        descr="Status",
        required=True,
        defaultValue="open",
        values={
            "open": "Offen",
            "pending": "In Bearbeitung",
            "closed": "Geschlossen",
        },
    )

    user = UserBone(
         descr="Zugewiesen an",
    )
```

## TodoSkel.sub_skel()

```py
TodoSkel.sub_skel("subject", "*name")
```

image

## TodoSkel.subSkel()

```py
skel = TodoSkel.subSkel("add")
```
image

**Comment by phorward** [comment]
SubSkels shall also be defined as bone parameters, so that the subSkels dict is not required to be set, but shall be supported backward compatible.

Additionally a bones parameter should be used instead of the tuple-condition.

```py
# legacy mode (see 2)
subskel = TodoSkel.subskel("add")
# creates subskel: key, firstname, lastname, subject

# free mode (see 3):
subskel = TodoSkel.subskel(bones=("subject", "message", "*stname"))
# creates subskel: key, subject, message, firstname, lastname

# mixed mode (see 4)
subskel = TodoSkel.subskel("add", bones=("message", ))
# creates subskel: key, firstname, lastname, subject, message
```

File: src/viur/core/skeleton.py [hunk 1]
```diff
@@ -137,38 +138,58 @@ class SkeletonInstance:
...
     }
 
-    def __init__(self, skelCls, subSkelNames=None, fullClone=False, clonedBoneMap=None):
-        if clonedBoneMap:
-            self.boneMap = clonedBoneMap
-            for k, v in self.boneMap.items():
-                v.isClonedInstance = True
+    def __init__(
+        self,
+        skel_cls: Skeleton,
```
Comment (sveneberth) [inline src/viur/core/skeleton.py:143]:
```suggestion
        skel_cls: t.Type[Skeleton],
```

File: src/viur/core/skeleton.py [hunk 2]
```diff
@@ -411,27 +432,88 @@ class BaseSkeleton(object, metaclass=MetaBaseSkel):
...
+        reason="Function renamed. Use subskel function as alternative implementation.",
+        action="always"
+    )
+    def subSkel(cls, *subskel_names, full_clone: bool = False, **kwargs) -> SkeletonInstance:
+        return cls.subskel(*subskel_names, full_clone=full_clone)  # FIXME: REMOVE WITH VIUR4
+
+    @classmethod
+    def subskel(
+        cls,
+        *names,
```
Comment (sveneberth) [inline src/viur/core/skeleton.py:446]:
```suggestion
        *names: str,
```

**Review by sveneberth — DISMISSED** [review]
works so far for me

**Pull request merged**